src/stavenote.ts

```diff
@@ -1263,7 +1263,8 @@ export class StaveNote extends StemmableNote {
 
     // Apply the overall style -- may be contradicted by local settings:
     this.applyStyle();
-    ctx.openGroup('stavenote', this.getAttribute('id'));
+    const cls = 'stavenote' + (this.getAttribute('class') ? ' ' + this.getAttribute('class') : '');
```

Comment:
You're calling this.getAttribute('class') twice here. This can be stored in a variable to reduce processing:
```js
const clsAttribute = this.getAttribute('class');
const cls = 'stavenote' + (clsAttribute ? ' ' + clsAttribute : '');
```
same for stem.ts line 214.